backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,301 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// New Backend
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "(Required) The address of the REST endpoint.",
+			},
+
+			"update_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "POST",
+				Description: "(Optional) HTTP method to use when updating state. Defaults to POST.",
+			},
+
+			"lock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the lock REST endpoint. Defaults to address.",
+			},
+
+			"lock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "LOCK",
+				Description: "(Optional) The HTTP method to use when locking. Defaults to LOCK.",
+			},
+
+			"unlock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the unlock REST endpoint. Defaults to address.",
+			},
+
+			"unlock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "UNLOCK",
+				Description: "(Optional) The HTTP method to use when unlocking. Defaults to UNLOCK.",
+			},
+
+			"username": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The username for HTTP basic authentication.",
+			},
+
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The password for HTTP basic authentication.",
+			},
+
+			"skip_cert_verification": {
```

Comment:
I would consider adding a `ConflictsWith` clause to this argument. `ConflictsWith` will ensure only one can be set, and will significantly reduce your verification logic configure.

`skip_cert_verfication` conflicts with `local_ca_cert_file`, `mutual_tls_authentication`